src/env.cc

```diff
@@ -151,6 +153,10 @@ void Environment::Start(int argc,
   set_process_object(process_object);
 
   SetupProcessObject(this, argc, argv, exec_argc, exec_argv);
+
+  // Used by EnvPromiseHook to know that we are on a node context.
+  context()->SetAlignedPointerInEmbedderData(kContextEmbedderDataIndex + 1, (void *)kNodeContextTag);
```

Comment:
1. We have an entire system worked out for embedder data index in [node\_context\_data.h](https://github.com/nodejs/node/blob/d1156da815d878df9252cd0b0489c27c38e63511/src/node_context_data.h). You should build on that.
2. Can we instead use an actual memory address to avoid clashing?

Reply:
sure, I can move it to use that header and all that jazz
what do you mean by 2?